### PR TITLE
IDE: Collect statistics when converting json to struct on paste

### PR DESCRIPTION
**Pull request by dima74**
Inspired by `GoGenerateTypeFromJsonUsageCollector`

Note, any statistics are collected only if `Send usage statistics` option in `Preferences | Appearance & Behavior | System Settings | Data Sharing` is enabled

See #7816

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package org.rust.ide.statistics
+
+import com.intellij.internal.statistic.eventLog.EventLogGroup
+import com.intellij.internal.statistic.eventLog.events.EventFields
+import com.intellij.internal.statistic.service.fus.collectors.CounterUsagesCollector
+
+@Suppress("UnstableApiUsage")
+object RsConvertJsonToStructUsagesCollector : CounterUsagesCollector() {
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt:13]: Actually, I'd merge all code from this class to `RsCounterUsagesCollector`
If it's ok for you, don't forget to provide more specific names for log methods

File: src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    override fun getGroup(): EventLogGroup = GROUP
+
+    private val GROUP = EventLogGroup("rust.generate.type.from.json.usage", 1)
+
+    private val JSON_LIKE_TEXT_PASTED = GROUP.registerEvent("json.like.text.pasted")
+    private val PASTED_JSON_CONVERTED = GROUP.registerEvent("pasted.json.converted")
+
+    private val REMEMBER_CHOICE_RESULT = GROUP.registerEvent(
+        "remember.choice.result",
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt:23]: nit: maybe introduce a common prefix for these events to simplify future search?

File: src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dima74) [reply src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt:13]: Isn't there some special handling for `object`s in the platform? There are such examples, e.g. `VcsCloneCollector`

File: src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt (hunk 2 above)
Comment (dima74) [reply src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt:23]: I guess "json.like.text.pasted" and "pasted.json.converted" are unique enough, will it be ok to replace just "remember.choice.result" to "json.paste.dialog.remember.choice.result"?

File: src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/ide/statistics/RsConvertJsonToStructUsagesCollector.kt:13]: If nothing changed recently, then no
But you can check it in debugger yourself

**Review by Undin — APPROVED** [review]
Looks ok except minor comment

**Comment by dima74** [comment]
bors r=Undin

**Pull request merged**